File: android/trunk/src/com/usedparts/domain/Order.java
package com.usedparts.domain;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.*;

public class Order {
    @JsonProperty("ix")
    public long id;

    @JsonProperty("head")
    public String name;

    @JsonProperty("body")
    public String body;

    @JsonProperty("offers")
    public int offersCount;

    @JsonProperty("advDate")
    @JsonFormat(shape=JsonFormat.Shape.STRING, pattern="dd.MM.yyyy")
    public Date posted;

    @JsonProperty("imgs")
    public List<String> images;

    @JsonProperty("thumbs")
    public List<String> thumbs;

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        } else if (obj instanceof Order) {
            Order order = (Order) obj;
            return order.id == this.id;
        } else {
            return false;
        }
    }
}

File: wp/trunk/UsedParts.Domain/Order.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using UsedParts.Domain.Converters;

namespace UsedParts.Domain
{
    public class Order
    {
        [JsonProperty(PropertyName = "ix")]
        public int Id { get; set; }

        [JsonProperty(PropertyName = "head")]
        public string Name { get; set; }

        [JsonProperty(PropertyName = "body")]
        public string Body { get; set; }

        [JsonProperty(PropertyName = "offers")]
        public int OffersCount { get; set; }

        [JsonProperty(PropertyName = "advDate")]
        [JsonConverter(typeof(DateConverter))]
        public DateTime Posted { get; set; }

        [JsonProperty(PropertyName = "imgs")]
        public IEnumerable<string> Images { get; set; }

        #region equality

        protected bool Equals(Order other)
        {
            return Id == other.Id;
        }

        public override int GetHashCode()
        {
            return Id;
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((Order) obj);
        }

        #endregion
    }
}
